crates/sui-json-rpc/src/read_api.rs

```diff
@@ -927,62 +926,30 @@ async fn get_display_fields(
 async fn get_display_object_by_type(
     fullnode_api: &ReadApi,
     object_type: &StructTag,
-) -> RpcResult<Option<DisplayObject>> {
-    let display_object_id = get_display_object_id(fullnode_api, object_type).await?;
-    if display_object_id.is_none() {
-        return Ok(None);
-    }
-    // safe to unwrap because `is_none` is checked above
-    let display_object_id = display_object_id.unwrap();
-    if let ObjectRead::Exists(_, display_object, _) = fullnode_api
+    // TODO: add query version support
+) -> RpcResult<Option<DisplayVersionUpdatedEvent>> {
+    let mut events = fullnode_api
         .state
-        .get_object_read(&display_object_id)
-        .await
-        .map_err(|e| anyhow!("Failed to fetch display object {display_object_id}: {e}"))?
-    {
-        let move_object = display_object
-            .data
-            .try_as_move()
-            .ok_or_else(|| anyhow!("Failed to extract Move object from {display_object_id}"))?;
-        Ok(Some(
-            bcs::from_bytes::<DisplayObject>(move_object.contents()).map_err(|e| {
-                anyhow!("Failed to deserialize DisplayObject {display_object_id}: {e}")
-            })?,
-        ))
+        .query_events(
+            EventFilter::MoveEventType(DisplayVersionUpdatedEvent::type_(object_type)),
+            None,
+            1,
+            true,
+        )
+        .await?;
+
+    // If there's any recent version of Display, give it to the client.
+    // TODO: add support for version query.
+    if let Some(event) = events.pop() {
+        let display: DisplayVersionUpdatedEvent = bcs::from_bytes(&event.bcs[..])
+            .map_err(|e| anyhow!("Failed to deserialize 'VersionUpdatedEvent': {e}"))?;
+
+        Ok(Some(display))
     } else {
-        Err(anyhow!("Display object {display_object_id} does not exist"))?
+        Err(anyhow!("Display object for {object_type} does not exist"))?
```

Comment:
I think the correct logic is to return `Ok(None)` if an event cannot be found. https://github.com/MystenLabs/sui/pull/9729

Reply:
Kk